src/Lykke.Service.Operations/Workflow/Sagas/BlockchainCashoutSaga.cs

```diff
@@ -106,12 +103,35 @@ public async Task Handle(SolarCashOutCompletedEvent evt, ICommandSender commandS
 
             commandSender.SendCommand(command, "operations");
         }
-        
+
+        #region CashourProcessorEvents
+
         [UsedImplicitly]
-        public async Task Handle(CashoutCompletedEvent evt, ICommandSender commandSender)
+        public async Task Handle(CashoutsBatchCompletedEvent evt, ICommandSender commandSender)
         {
-            _log.Info($"CashoutCompletedEvent for BIL received. Operation [{evt.OperationId}]", evt);
+            if (evt.Cashouts == null || evt.Cashouts.Length == 0)
+            {
+                throw new InvalidOperationException($"Empty cashouts in batch. BatchId [{evt.BatchId}]");
```

Comment:
Why throwing exception? Why this is a problem? What the service can do by retrying the event? The same applies to the code below.